platform/darwin/src/NSPredicate+MLNAdditions.mm

```diff
@@ -73,6 +73,15 @@ + (instancetype)predicateWithMLNJSONObject:(id)object {
     if (operatorTypeNumber) {
         NSPredicateOperatorType operatorType = (NSPredicateOperatorType)[operatorTypeNumber unsignedIntegerValue];
         
+        // Handle ["in", "attributeName", "val"..] -> ["in", ["get", "attributeName"], ["literal", ["val"..]]]
```

Comment:
This is not consistent with the MapLibre Style Spec. https://maplibre.org/maplibre-style-spec/expressions/#in

Reply:
Good point. Looks like there is a practice that I have been using and that are used in some examples that are according to the documentation.

Here are some examples from metrics/integration/styles/chicago.json of this practice. Not according to documentation, but works in .js and native, but not in `predicateWithMLNJSONObject`.
https://github.com/maplibre/maplibre-native/blob/c3a2fa247cd7dd15742919c537152d649fa9e31d/metrics/integration/styles/chicago.json#L68-L73
https://github.com/maplibre/maplibre-native/blob/c3a2fa247cd7dd15742919c537152d649fa9e31d/metrics/integration/styles/chicago.json#L128-L139